model/k8s/metrics.yaml

```
@@ -616,3 +616,341 @@ groups:
+      The value represents the configured quota limit of the resource in the namespace.
+    instrument: updowncounter
+    unit: "By"
+    entity_associations:
+      - k8s.resourcequota
+    note: |
+      This metric is retrieved from the `hard` field of the
+      [K8s ResourceQuotaStatus](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#resourcequotastatus-v1-core).
+  - id: metric.k8s.resourcequota.memory.limit.used
+    type: metric
+    metric_name: k8s.resourcequota.memory.limit.used
+    stability: development
+    brief: |
+      The memory limits in a specific namespace.
+      The value represents the current observed total usage of the resource in the namespace.
+    instrument: updowncounter
+    unit: "By"
+    entity_associations:
+      - k8s.resourcequota
+    note: |
+      This metric is retrieved from the `used` field of the
+      [K8s ResourceQuotaStatus](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#resourcequotastatus-v1-core).
+  - id: metric.k8s.resourcequota.memory.request.hard
+    type: metric
+    metric_name: k8s.resourcequota.memory.request.hard
+    stability: development
+    brief: |
+      The memory requests in a specific namespace.
+      The value represents the configured quota limit of the resource in the namespace.
+    instrument: updowncounter
+    unit: "By"
+    entity_associations:
+      - k8s.resourcequota
+    note: |
+      This metric is retrieved from the `hard` field of the
+      [K8s ResourceQuotaStatus](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#resourcequotastatus-v1-core).
+  - id: metric.k8s.resourcequota.memory.request.used
+    type: metric
+    metric_name: k8s.resourcequota.memory.request.used
+    stability: development
+    brief: |
+      The memory requests in a specific namespace.
+      The value represents the current observed total usage of the resource in the namespace.
+    instrument: updowncounter
+    unit: "By"
+    entity_associations:
+      - k8s.resourcequota
+    note: |
+      This metric is retrieved from the `used` field of the
+      [K8s ResourceQuotaStatus](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#resourcequotastatus-v1-core).
+
+  # k8s.resourcequota.hugepages.* metrics
+  - id: metric.k8s.resourcequota.hugepages.request.hard
+    type: metric
+    metric_name: k8s.resourcequota.hugepages.request.hard
+    stability: development
+    brief: |
+      The huge page requests in a specific namespace.
+      The value represents the configured quota limit of the resource in the namespace.
+    instrument: updowncounter
+    unit: "By"
+    entity_associations:
+      - k8s.resourcequota
+    note: |
+      This metric is retrieved from the `hard` field of the
+      [K8s ResourceQuotaStatus](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#resourcequotastatus-v1-core).
+    attributes:
+      - ref: k8s.hugepage.size
```

```suggestion
      - ref: k8s.hugepages.size
```
Wdyt think about using the plural form consistently? K8s spec uses `hugepages` and our metrics also have the plural form.

---

I think we should avoid pluralisation here. `size` is an attribute of a `hugepage` so `hugepages.size` looks a bit weird.
For consistency, I think that the respective metrics should not be pluralised but use `count` instead so as to be aligned with `k8s.resourcequota.object_count.*`.
I suggest we have them as `k8s.resourcequota.hugepage_count.*` and `k8s.resourcequota.persistentvolumeclaim_count.*` accordingly. WDYT?

Tried that at [0eacaa7](https://github.com/open-telemetry/semantic-conventions/pull/2113/commits/0eacaa7f084b92e362d223d6397c042b953fe19d).